doc/HACKING/abbrevs-proposed.txt

```diff
@@ -0,0 +1,890 @@
+        cdm
+
+constant:
+        const
+
+context:
+        ctx
+
+control, controller:
+        ???
+        ctl # proposed: standardize on this?
+
+count:
+        cnt
+
+counter:
+        ctr
+
+country code:
+        cc
+
+credential:
+        cred
+
+cryptographic, cryptography:
+        crypt  # can mostly omit?
+        crypto # can mostly omit?
+
+cumulative distribution function:
+        cdfs
+
+current:
+        cur   # prefer
+        curr  # deprecate
+
+data-independent:
+        di
+
+database:
+        db
+
+declaration, declare:
+        decl
+
+decode, decrypt:
+        dec
+
+decrease, decrement:
+        dec
+
+default:
+        dflt
+
+define:
+        def
+
+delete:
+        del
+
+delivery, delivery:
+        deliv # proposed
+
+denial of service:
+        dos
+
+describe, description:
+        desc  # deprecate
+        descr # proposed
+
+descriptor:
+        desc
+
+destination:
+        dest
+
+destroy:
+        ???
+
+detach:
+        ???
+
+deterministic:
+        determ # proposed
+
+difference:
+        diff
+
+diffie-hellman:
+        dh
+
+digest map:
+        dmap
+
+directory:
+        d
+        dir # preferred
+
+distribution, distributor:
+        dist # preferred
+        distr
+
+distributor:
+        dist
+
+double digest:
+        dd
+
+download:
+        dl
+
+duplicate:
+        dup
+
+dynamic:
+        dyn
+
+ed25519:
+        ed
+
+element:
+        elt
+
+elliptic curve:
+        ec
+
+encode, encrypt:
+        enc
+
+encrypted:
+        ???
+
+end-of-file:
+        eof
+
+end-of-string:
+        eos
+
+end-to-end:
+        e2e
+
+entry:
+        ent
+
+environment:
+        env
+
+equals:
+        eq
+
+error:
+        err
+
+escape:
+        esc
+
+escaped:
+        ??? # proposed: no abbreviation
+
+establish, established:
+        estab # proposed
+
+event:
+        ev
+        evt (rare) # deprecate
+
+exclude, exclusive:
+        excl
+
+execute:
+        exec
+
+expand, expire:
+        exp
+
+exponentially weighted moving average:
+        ewma
+
+extend:
+        ext??? # deprecate; too ambiguous
+
+extension:
+        ext
+
+extrainfo:
+        ei
+
+field:
+        ??? # proposed: no abbreviation
+
+file:
+        f # proposed: deprecate
+
+file descriptor:
+        fd
+
+filename:
+        fn # deprecate
+        fname
+
+filter:
+        filt
+
+final, finalize, finish:
+        fin  # preferred
+        fini # deprecate
+
+fingerprint:
+        fp
+
+format:
+        fmt
+
+forward:
+        fwd
+
+fraction:
+        frac
+
+frequency:
+        freq # proposed
+
+function:
+        fn   # preferred
+        func # deprecate
+
+greater than:
+        gt
+
+greatest common devisor:
+        gcd
+
+handle:
+        hnd # proposed
+
+handshake:
+        ??? # proposed: no abbreviation
+
+hash table:
+        ht
+
+hexadecimal:
+        hex
+
+hibernate, hibernating, hibernation:
+        ??? # proposed: rename entirely; it gets confused with dormancy.
+
+hidden service:
+        hs
```

Comment:
Yes, I think this is an important missing piece. Is it fine to still abbreviate 'onion service' as 'hs'? Because we are clearly doing it.

Reply:
I honestly don't mind.

I think "hs" is a fine abbreviation for code and names that users never see.
The abbreviations for onion service are all longer, because "os" means "operating system".